src/main/java/com/uwetrottmann/tmdb2/TmdbInterceptor.java

```diff
@@ -39,10 +70,95 @@ public static Response handleIntercept(Chain chain, String apiKey) throws IOExce
         HttpUrl.Builder urlBuilder = request.url().newBuilder();
         urlBuilder.setEncodedQueryParameter(Tmdb.PARAM_API_KEY, apiKey);
 
+        if (request.url().pathSegments().get(0).equals("account")) {
+            if (tmdb.isLoggedIn() && tmdb.hasAccountSession()) {
+                urlBuilder.addEncodedQueryParameter(Tmdb.PARAM_SESSION_ID, tmdb.getSessionId());
+            } else {
+                throw new TmdbUnathorizedException("You do not have permissions to access the service.");
+            }
+        } else if (request.url().pathSegments().get(0).equals("guest_session")) {
+            if (tmdb.isLoggedIn() && tmdb.hasGuestSession()) {
+                urlBuilder.addEncodedQueryParameter(Tmdb.PARAM_GUEST_SESSION_ID, tmdb.getGuestSessionId());
+            } else {
+                throw new TmdbUnathorizedException("You do not have permissions to access the service.");
+            }
+        } else if (request.url().pathSegments().get(request.url().pathSegments().size()-1).equals("account_states")) {
+            if (tmdb.isLoggedIn() && tmdb.hasGuestSession()) {
+                urlBuilder.addEncodedQueryParameter(Tmdb.PARAM_GUEST_SESSION_ID, tmdb.getGuestSessionId());
+            } else {
+                throw new TmdbUnathorizedException("You do not have permissions to access the service.");
+            }
+        } else if (!request.method().toLowerCase().equals("get")) {
+            if (tmdb.isLoggedIn() && tmdb.hasAccountSession()) {
+                urlBuilder.addEncodedQueryParameter(Tmdb.PARAM_SESSION_ID, tmdb.getSessionId());
+            } else if (
+                    tmdb.isLoggedIn()
+                            && tmdb.hasGuestSession()
+                            && request.url().pathSegments().get(request.url().pathSegments().size() - 1).equals("rating")
+                    ) {
+                urlBuilder.addEncodedQueryParameter(Tmdb.PARAM_GUEST_SESSION_ID, tmdb.getGuestSessionId());
+            } else {
+                throw new TmdbUnathorizedException("You do not have permissions to access the service.");
+            }
+        }
+
         Request.Builder builder = request.newBuilder();
         builder.url(urlBuilder.build());
+        builder.header("Content-Type", "application/json;charset=utf-8");
```

Comment:
I'll have a look at the test method output once some POST methods are available. I believe @SimonVT is right in that the header is added automatically by retrofit/gson.